server/Client.cc

```diff
@@ -1291,6 +1318,52 @@ Client::get_files(DBus::Connection& conn, DBus::Message& msg)
 }
 
 
+struct PathException : public std::exception
+{
+    explicit PathException() throw() {}
+    virtual const char* what() const throw() { return "Invalid path"; }
+};
+
+
+void
+Client::get_files_pipe(DBus::Connection& conn, DBus::Message& msg)
+{
+    string config_name;
+    string path; // relative to volume root
+    dbus_uint32_t num1, num2;
+
+    DBus::Hihi hihi(msg);
+    hihi >> config_name >> path >> num1 >> num2;
```

Comment:
How is the path parameter to be used? Shouldn't there also be a depth parameter?

E.g. then the client could request the files of the root with path=/ and depth=1. Later the client could request the files one level below.